webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx

```diff
@@ -334,15 +195,15 @@ const ExperimentTrials: React.FC<Props> = ({ experiment }: Props) => {
       });
       setIsLoading(false);
     }
-  }, [ experiment.id, canceler, pagination, sorter, filters ]);
+  }, [ experiment.id, canceler, settings ]);
```

Comment:
Nice catches on the dependency arrays! This one though (`fetchExperimentTrials`) I think is still needed right? Since it references `settings.tableLimit|tableOffset|sortDesc|sortKey|state` etc...

Reply:
Yes, but here (and below), we shouldn't refetch data when `settings.compare` changes right? If you just put `settings` in the dependency array it will update the callback function and also fetch when any part of `settings` changes, which includes `settings.compare`.

Reply:
ah I see, you're suggesting that we should spell out the exact ones we depend on rather than just the entire settings object change.

Reply:
In this case yes, I think that might be best. It makes the code a little less clean but should reduce API calls.